storage/volume/core.go

```diff
@@ -159,43 +159,43 @@ func (c *Core) getVolumeDriver(id types.VolumeID) (*types.Volume, driver.Driver,
 }
 
 // existVolume return 'true' if volume be found and not errors.
-func (c *Core) existVolume(id types.VolumeID) (bool, error) {
-	_, err := c.getVolume(id)
+func (c *Core) existVolume(id types.VolumeContext) (*types.Volume, bool, error) {
```

Comment:
I am wondering if it is necessary to update the return parameter of this function. I saw that you added an instance of `*types.Volume` in the function.

When I searched the code, I found that this is used in `core.CreateVolume`, and this `*types.Volume` is only used when the input volume ID exists, then you returned the volume.

I think with the change, you break the primitives of this function.

As an encapsulated package, we should make the function to be focused as much as possible. when checking if it exists, the main part is the returned boolean, if the returned error is necessary, it is ok. 

But when we return a Volume instance, it combines two actions into a single one, to break the purity of this function `existVolume`.

If the caller need the `types.Volume` instance when it is already exists, please let it use another function to get it, like `core.GetVolumes(...)`.

WDYT? @rudyfly 

In addition, would you mind changing function name into `isVolumeExist`?

Reply:
good idea, I fix it.